testing/src/test/java/io/stargate/it/MultipleStargateInstancesTest.java

```diff
@@ -57,26 +59,28 @@ public MultipleStargateInstancesTest(ClusterConnectionInfo backend) {
 
   @BeforeEach
   public void setup(TestInfo testInfo) {
-    DriverConfigLoader loader =
-        DriverConfigLoader.programmaticBuilder()
-            .withBoolean(DefaultDriverOption.METADATA_TOKEN_MAP_ENABLED, false)
-            .withString(
-                DefaultDriverOption.LOAD_BALANCING_POLICY_CLASS,
-                DcInferringLoadBalancingPolicy.class.getName())
-            .withDuration(DefaultDriverOption.REQUEST_TIMEOUT, Duration.ofSeconds(30))
-            .withDuration(DefaultDriverOption.CONNECTION_INIT_QUERY_TIMEOUT, Duration.ofSeconds(30))
-            .withDuration(DefaultDriverOption.CONTROL_CONNECTION_TIMEOUT, Duration.ofSeconds(30))
-            .withDuration(DefaultDriverOption.REQUEST_TRACE_INTERVAL, Duration.ofSeconds(1))
-            .withStringList(
-                DefaultDriverOption.METRICS_NODE_ENABLED,
-                Collections.singletonList(DefaultNodeMetric.CQL_MESSAGES.getPath()))
-            .build();
-
-    CqlSessionBuilder cqlSessionBuilder = CqlSession.builder().withConfigLoader(loader);
+    OptionsMap config = OptionsMap.driverDefaults();
+    config.put(TypedDriverOption.METADATA_TOKEN_MAP_ENABLED, false);
+    config.put(
+        TypedDriverOption.LOAD_BALANCING_POLICY_CLASS,
+        DcInferringLoadBalancingPolicy.class.getName());
+    config.put(TypedDriverOption.REQUEST_TIMEOUT, Duration.ofSeconds(30));
+    config.put(TypedDriverOption.CONNECTION_INIT_QUERY_TIMEOUT, Duration.ofSeconds(30));
+    config.put(TypedDriverOption.CONTROL_CONNECTION_TIMEOUT, Duration.ofSeconds(30));
+    config.put(TypedDriverOption.REQUEST_TRACE_INTERVAL, Duration.ofSeconds(1));
+    config.put(
+        TypedDriverOption.METRICS_NODE_ENABLED,
+        Collections.singletonList(DefaultNodeMetric.CQL_MESSAGES.getPath()));
+
+    CqlSessionBuilder cqlSessionBuilder =
+        CqlSession.builder().withConfigLoader(DriverConfigLoader.fromMap(config));
     for (String host : getStargateHosts()) {
       cqlSessionBuilder.addContactPoint(new InetSocketAddress(host, 9043));
     }
     session = cqlSessionBuilder.build();
+    await()
+        .atMost(Duration.ofMinutes(3))
```

Comment:
I'd go with a pessimistic timeout, something like 10 min, perhaps? Increasing the timeout will not actually increase the run time of successful (expected) runs, but will reduce spurious failures as we know CI is very slow.

Reply:
I think this timeout actually happened in the first CI run on this PR :thinking:

Reply:
Yes, didn't see your comment but it was reached and I did increase it to 10 minutes.
Also added logs to check whether `system.peers` is not updated or the driver didn't get a gossip event.